DotNetDBF/DBFReader.cs

```diff
@@ -479,5 +479,39 @@ internal object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
 
             return selectIndexes.Any() ? selectIndexes.Select(it => recordObjects[it]).ToArray() : recordObjects;
         }
+	
+	/// <summary>
+        /// Reads the data table from DBF from current position to end.
+        /// </summary>
+        /// <returns>The data table with all data, original types and names of columns.</returns>
+        /// <param name="dataTableName">Name for data table.</param>
+        public DataTable ReadDataTable(string dataTableName)
+        {
+            DataTable DT = new DataTable(dataTableName);
+            for (var i = 0; i < _header.FieldArray.Length; i++)
+            {
+                DBFField DF = _header.FieldArray[i];
+                string n = DF.Name;
+                DT.Columns.Add(n, DF.Type);
+                if (DF.Type == typeof(System.String))
+                    DT.Columns[n].MaxLength = DF.FieldLength;
+            }

```

Comment:
Think i would make more sense if `_selectFields`, `_orderedSelectFields` was taken into account to describe the table.

Reply:
Thanks!  `foreach (int i in _orderedSelectFields)`is better.